solidity/test/token_stake/TestStakingPortBacker.js

```diff
@@ -0,0 +1,635 @@
+      const data = web3.eth.abi.encodeParameters(["address"], [operatorFour])
+      await expectRevert(
+        token.approveAndCall(
+          stakingPortBacker.address, delegatedAmount, data, {from: tokenOwner}
+        ),
+        "Stake not copied for the operator"
+      )  
+    })
+
+    it("fails for corrupted input data", async () => {
+      await expectRevert(
+        token.approveAndCall(
+          stakingPortBacker.address, 
+          delegatedAmount, 
+          web3.eth.abi.encodeParameters(["uint256", "uint256"], [1, 2]), 
+          {from: tokenOwner}
+        ),
+        "Corrupted input data"
+      )    
+    })
+
+    it("can be done only one time for the operator", async () => {
+      const data = web3.eth.abi.encodeParameters(["address"], [operatorOne])
+      await token.approveAndCall(
+        stakingPortBacker.address, delegatedAmount, data, {from: tokenOwner}
+      )  
+      await expectRevert(
+        token.approveAndCall(
+          stakingPortBacker.address, delegatedAmount, data, {from: tokenOwner}
+        ),
+        "Already paid back"
+      ) 
+    })
+
+    it("changes liquid tokens staking relationship owner to token owner", async () => {
+      const data = web3.eth.abi.encodeParameters(["address"], [operatorOne])
+      await token.approveAndCall(
+        stakingPortBacker.address, delegatedAmount, data, {from: tokenOwner}
+      )
+      
+      expect(await newTokenStaking.ownerOf(operatorOne)).to.equal(tokenOwner)
+    })
+
+    it("changes grant staking relationship owner to grantee", async () => {
+      const data = web3.eth.abi.encodeParameters(["address"], [operatorTwo])
+      await token.transfer(grantee, delegatedAmount, {from: deployer})
+      await token.approveAndCall(
+        stakingPortBacker.address, delegatedAmount, data, {from: grantee}
+      )
+      
+      expect(await newTokenStaking.ownerOf(operatorTwo)).to.equal(grantee) 
+    })
+
+    it("changes managed grant staking relationship owner to managed grantee", async () => {
+      const data = web3.eth.abi.encodeParameters(["address"], [operatorThree])
+      await token.transfer(managedGrantee, delegatedAmount, {from: deployer})
+      await token.approveAndCall(
+        stakingPortBacker.address, delegatedAmount, data, {from: managedGrantee}
+      )
+      
+      expect(await newTokenStaking.ownerOf(operatorThree)).to.equal(managedGrantee) 
+    })
+
+    it("fails when not done by the eventual staking relationship owner", async () => {
+      const data = web3.eth.abi.encodeParameters(["address"], [operatorOne])
+      await expectRevert(
+        token.approveAndCall(
+          stakingPortBacker.address, delegatedAmount, data, {from: operatorOne}
+        ),
+        "Not authorized to pay back"
+      )
+      await expectRevert(
+        token.approveAndCall(
+          stakingPortBacker.address, delegatedAmount, data, {from: thirdParty}
+        ),
+        "Not authorized to pay back"
+      )
+    })
```

Comment:
What should happen when `receiveApproval` is called directly (not via `token.approveAndCall`)? Do we want to add such test case?

Reply:
I don't think it's a problem with this contract, I don't see any scenario for exploit here. 🤔 I would leave it as-is, for the sake of less complexity.